### PR TITLE
Python gateway & webclient delete improvements

### PR DESCRIPTION
**Pull request by ximenesuk**
This PR addresses the various items in https://trello.com/c/WMnOBpC6/31-improve-deleteobjects-deprecate-deleteobjectdirect-inpython-gateway both RFEs and deprecation of `deleteObjectDirect`.

Tests, both `OmeroPy` and `OmeroWeb` should pass.

Web functionality to be tested is that which deletes and unlinks P/D/I, S/P and Annotations. I.e quite wide-ranging, but concentrating on unlinking and deleting all types of annotations would be useful.

### CODE REVIEW

File: components/tools/OmeroWeb/omeroweb/webclient/controller/container.py [hunk 1]
```diff
@@ -1407,8 +1431,8 @@ def copyImageToDataset(self, source, destination=None):
         if destination is None:
```
Comment (ximenesuk) [inline components/tools/OmeroWeb/omeroweb/webclient/controller/container.py]: I can't see any direct use of this method. Should it be removed?

File: components/tools/OmeroWeb/omeroweb/webclient/controller/container.py [hunk 2]
```diff
@@ -1323,7 +1346,8 @@ def removemany(self, images):
         if self.dataset is not None:
```
Comment (ximenesuk) [inline components/tools/OmeroWeb/omeroweb/webclient/controller/container.py]: I can't see any direct use of this method. Should it be removed?

**Comment by ximenesuk** [comment]
Closing temporarily

**Comment by ximenesuk** [comment]
`DeleteObjects()` doesn't take full advantage of the newer functionalities in `Delete2` such as deleting different objects in one command. At some point a useful change might be to add something like a `DeleteMultipleObjects()` which takes a map of object types and ids.

**Comment by will-moore** [comment]
Code changes seem to make sense on overview. Haven't tested functionality yet.

**Comment by ximenesuk** [comment]
@sbesson the one thing I haven't done is mark `deleteObjectsDirect()` as deprecated. I was trying to recall if you had any luck with this or whether just adding a comment would have the same effect.

**Comment by joshmoore** [comment]
A vote for `DeprecationWarning`:

```
$ git grep -l Deprec | grep Warning
# joshuas-mbp:ome0 jamoore$ git grep -n Deprec | grep Warning
components/tools/OmeroPy/src/omero/__init__.py:129:        category=DeprecationWarning)
components/tools/OmeroPy/src/omero/gateway/__init__.py:2767:            DeprecationWarning)
components/tools/OmeroPy/src/omero/gateway/__init__.py:8873:            "Deprecated. Use getImportedImageFiles()", DeprecationWarning)
components/tools/OmeroPy/src/omero/plugins/sessions.py:169:                    " instead.", DeprecationWarning)
components/tools/OmeroPy/src/omero/plugins/sessions.py:176:                    " instead.", DeprecationWarning)
components/tools/OmeroPy/src/omero/util/script_utils.py:1225:            "future versions of OMERO", DeprecationWarning)
components/tools/OmeroPy/src/omero/util/temp_files.py:140:                DeprecationWarning)
```

**Comment by ximenesuk** [comment]
@joshmoore thanks. Clearly my `git grep`ing isn't up to scratch! I'll formally deprecate the method.

**Comment by ximenesuk** [comment]
In looking through the gateway I noticed that:
https://github.com/ximenesuk/openmicroscopy/blob/python-delete-improvements/components/tools/OmeroPy/src/omero/gateway/__init__.py#L7129
has been deprecated since 4.3.2, this would probably be a good time to remove this method.

**Comment by will-moore** [comment]
Functionality of removing annotations in webclient works fine.
```container.py``` certainly has some code that's not used now since the jsTree work, including ```move(), removemany() and copyImageToDataset()``` This will be removed as part of a full clean-up of this for 5.2.1.
Good to merge.

**Comment by joshmoore** [comment]
If they will be removed soon, deprecating sooner would be appropriate.

**Comment by will-moore** [comment]
@joshmoore I don't think that anything in the webclient can be considered an API (not like webgateway) - We don't deprecate or document changes to methods in webclient, we just go ahead and edit.

**Comment by joshmoore** [comment]
Longer-term we should really re-evaluate that since it increases the burden on consumers. Or at least it needs to be made completely obvious, e.g. `s/webclient/private/`

**Comment by sbesson** [comment]
The `omeroweb.webclient` module has certainly beenexposed in the [public Python API](http://downloads.openmicroscopy.org/omero/5.1.4/api/python/omeroweb/omeroweb.webclient.html) for every single release as well. I would vote for dropping the ambiguity as soon as possible.

**Comment by sbesson** [comment]
See failure in https://ci.openmicroscopy.org/job/OMERO-DEV-merge-integration-python/lastCompletedBuild/testReport/OmeroPy.test.integration.gatewaytest.test_get_objects/TestGetObject/testTraversal/

**Comment by will-moore** [comment]
@joshmoore @sbesson I would like to move all the "api" urls and view methods that power the jsTree etc into a new /url/ app, and add similar methods for other parts of the app as we go. Then mark everything remaining in webclient as non-public.

**Comment by ximenesuk** [comment]
I hadn't realised that the `getDataset` was still being tested despite being deprecated since 4.3.2. Its use in the test has been removed in this final commit.

**Comment by ximenesuk** [comment]
See https://ci.openmicroscopy.org/view/Failing/job/OMERO-DEV-merge-integration-python/110/

**Comment by sbesson** [comment]
Given the state of https://ci.openmicroscopy.org/view/DEV/job/OMERO-DEV-merge-integration-python/ and https://ci.openmicroscopy.org/view/DEV/job/OMERO-DEV-merge-integration-Python27/, merging this. Let's have the webclient discussion separately. @will-moore: can you provide a listing of what is meant to be public and private within webclient?

**Comment by will-moore** [comment]
Currently @dpwrussell added a number of 'api' urls, that map to corresponding functions in views.py and tree.py that load the data via projection queries and marshal it to json:
```
api/groups/
api/experimenters/
api/experimenters/
api/containers/
api/datasets/
api/images/
api/share_images/
api/plates/
api/plate_acquisitions/
api/links/
api/paths_to_object/
api/tags/
api/shares/
```
These make a good starting point for public methods in webclient. More of what the webclient needs should be moved to this architecture and made public, probably in a new 'api' Django app.

**Pull request merged**